src/qibocal/update.py

```diff
@@ -0,0 +1,151 @@
+
+
+def threshold(threshold: float, platform: Platform, qubit: QubitId):
+    platform.qubits[qubit].threshold = float(threshold)
+
+
+def mean_gnd_states(gnd_state: list, platform: Platform, qubit: QubitId):
+    """Update mean ground state value in platform for specific qubit."""
+    platform.qubits[qubit].mean_gnd_states = [float(state) for state in gnd_state]
+
+
+def mean_exc_states(exc_state: list, platform: Platform, qubit: QubitId):
+    """Update mean excited state value in platform for specific qubit."""
+    platform.qubits[qubit].mean_exc_states = [float(state) for state in exc_state]
+
+
+def classifiers_hpars(hpars: list, platform: Platform, qubit: QubitId):
+    """Update classifier hyperparameters in platform for specific qubit."""
+    platform.qubits[qubit].classifiers_hpars = hpars
+
+
+def virtual_phases(phases: dict[QubitId, float], platform: Platform, pair: QubitPairId):
+    """Update virtual phases for given qubits in pair in results."""
+    virtual_z_pulses = {
+        pulse.qubit.name: pulse
+        for pulse in platform.pairs[pair].native_gates.CZ.pulses
+        if isinstance(pulse, VirtualZPulse)
+    }
+    for qubit_id, phase in phases.items():
+        if qubit_id in virtual_z_pulses:
+            virtual_z_pulses[qubit_id].phase = phase
+        else:
+            virtual_z_pulses[qubit_id] = VirtualZPulse(
+                phase=phase, qubit=platform.qubits[qubit_id]
+            )
+            platform.pairs[pair].native_gates.CZ.pulses.append(
+                virtual_z_pulses[qubit_id]
+            )
+
+
+def CZ_duration(duration: int, platform: Platform, pair: QubitPairId):
+    """Update CZ duration for specific pair."""
+    for pulse in platform.pairs[pair].native_gates.CZ.pulses:
+        if pulse.qubit.name == pair[1]:
+            pulse.duration = int(duration)
+
+
+def CZ_amplitude(amp: float, platform: Platform, pair: QubitPairId):
+    """Update CZ amplitude for specific pair."""
+    for pulse in platform.pairs[pair].native_gates.CZ.pulses:
+        if pulse.qubit.name == pair[1]:
+            pulse.amplitude = float(amp)
+
+
+def t1(t1: int, platform: Platform, qubit: QubitId):
+    """Update mean excited state value in platform for specific qubit."""
+    platform.qubits[qubit].t1 = int(t1)
+
+
+def t2(t2: int, platform: Platform, qubit: QubitId):
+    """Update mean excited state value in platform for specific qubit."""
+    platform.qubits[qubit].t2 = int(t2)
+
+
+def t2_spin_echo(t2_spin_echo: float, platform: Platform, qubit: QubitId):
+    """Update mean excited state value in platform for specific qubit."""
+    platform.qubits[qubit].t2_spin_echo = int(t2_spin_echo)
+
+
+def drag_pulse_beta(beta: float, platform: Platform, qubit: QubitId):
+    """Update beta parameter e value in platform for specific qubit."""
+    shape = platform.qubits[qubit].native_gates.RX.shape
+    rel_sigma = re.findall(r"[\d]+[.\d]+|[\d]*[.][\d]+|[\d]+", shape)[0]
```

Comment:
I have not tested but this can probably be simplified using
```suggestion
    pulse = platform.qubits[qubit].native_gates.RX.pulse(start=0)
    rel_sigma = pulse.shape.rel_sigma
```
This will basically delegate the shenanigans to [qibolab's pulse constructor](https://github.com/qiboteam/qibolab/blob/352a4691a503b394cd12f57e455e6cddaed7aa10/src/qibolab/pulses.py#L951), but better to have them in a single place instead of two.

Reply:
Thanks @stavros11, I'll give it a try.